commands/login.go

```diff
@@ -221,16 +248,42 @@ func login(ctx context.Context, provider client.OpenIdProvider) (*LoginCmd, erro
 		return nil, fmt.Errorf("failed to generate SSH cert: %w", err)
 	}
 
-	// Write ssh secret key and public key to filesystem
-	if err := writeKeysToSSHDir(seckeySshPem, certBytes); err != nil {
-		return nil, fmt.Errorf("failed to write SSH keys to filesystem: %w", err)
-	}
-
 	idStr, err := IdentityString(*pkt)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse ID Token: %w", err)
 	}
-	fmt.Printf("Keys generated for identity\n%s\n", idStr)
+
+	if _, ok := HasSSHAgent(); ok {
+		pubkey, _, _, _, err := ssh.ParseAuthorizedKey(certBytes)
+		if err != nil {
+			return nil, err
+		}
+		cert, ok := pubkey.(*ssh.Certificate)
+		if !ok {
+			return nil, fmt.Errorf("failed to cast to certificate")
+		}
+
+		lifetime, err := getExpiration(pkt.Payload)
```

Comment:
This expiration is a little tricky because ID Tokens expire pretty quickly (~1 hour). So by default we don't use the ID Token expiry, but instead use IssuedAt + 24hours. However the server can be configured to use other options, including the exp claim and we don't know what expiry the server will use.

Unless you see a compelling reason otherwise, it is probably simplest to set this to IssuedAt + 23 hours.

Reply:
Ah, this makes sense.

But are we getting a new certificate every hour?

Also, would it make sense to include this expiry into the certificate `ValidBefore`/`ValidAfter` as well?

Reply:
>  I think for now we should just set 23 hour expiry for certs when not running in --auto-refresh and then solve this problem more completely when opkssh can act as the ssh agent. Do you think that is reasonable? If we go down this path we should make sure to document it in docs/sshagent.md

I think that is reasonable.

> ValidAfter might help. It wouldn't provide security since non of the fields in the cert are authenticated other than the public key, but may caught bugs and result better error messages if the client's clock is wrong.

We can do that in another PR then.

Reply:
Took at stab at this with https://github.com/openpubkey/opkssh/pull/96/commits/5e527b354ee85acb4c84baee151284400e88eb22